src/libs/actions/Report.js

```diff
@@ -336,6 +337,11 @@ function fetchChatReportsByIDs(chatList, shouldRedirectIfInaccessible = false) {
             Log.info('[Report] successfully fetched report data', false, {chatList});
             fetchedReports = reportSummaryList;
 
+            // Fetch last viewed report because we want to get its informations as soon as possible.
+            const earlyReport = _.find(fetchedReports, report => report.reportID === lastViewedReportID);
+            const simplifiedEarlyReport = getSimplifiedReportObject(earlyReport);
+            Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${earlyReport.reportID}`, simplifiedEarlyReport);
+
             // If we receive a 404 response while fetching a single report, treat that report as inaccessible.
             if (jsonCode === 404 && shouldRedirectIfInaccessible) {
```

Comment:
This will not work all the time. Check the next if condition it seems that if the report is not found, your code will break.

Reply:
yeah, my idea here was to load the report data of the opened report as soon as possible, but you're right, maybe we should load it with another way.